cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1225,6 +1226,85 @@ public void getInstances(FullHttpRequest request, HttpResponder responder,
     responder.sendJson(HttpResponseStatus.OK, GSON.toJson(output));
   }
 
+  /**
+   * Returns the run counts for all program runnables that are passed into the data. The data is an array of
+   * Json objects where each object must contain the following three elements: appId, programType, and programId.
+   * <p>
+   * Example input:
+   * <pre><code>
+   * [{"appId": "App1", "programType": "Service", "programId": "Service1"},
+   *  {"appId": "App1", "programType": "Workflow", "programId": "testWorkflow"},
+   *  {"appId": "App2", "programType": "Workflow", "programId": "DataPipelineWorkflow"}]
+   * </code></pre>
+   * </p><p>
+   * </p><p>
+   * The response will be an array of JsonObjects each of which will contain the three input parameters
+   * as well as 2 fields, "count" which maps to the count of the program and "statusCode" which maps to the
```

Comment:
nit: `count of the program runs`

Reply:
Better not to use `statusCode` as it is HTTP specific and confusing (as a HTTP response should only have one status code).

You can also choose to do it in a simpler way by just returning non-200 response status (i.e. it's either all succeeded or none).

Reply:
this is also the behavior I followed for our other batch calls(/instances, /status, etc), is it better to be consistent?

Reply:
To confirm if all succeeded or none then UI either has runs count of all the pipelines it shows or none.. Don't know if that would be desirable. For instance we have `statusCode` in the responses from wrangler. Something to consider.